### PR TITLE
Copying a PDS into existing PDS w/ no members

### PR DESCRIPTION
**Pull request by pujal0909**
**What It Does**

The copy command now recognizes partitioned data sets. It can copy members of a source PDS into an existing target PDS w/ no members and displays an error if the source PDS has 0 members.
https://github.com/zowe/zowe-cli/issues/2349

**How to Test**

Run the copy data set command with partitioned data sets.
zowe zos-files copy data-set "test-pds1" "test-pds2"

**Review Checklist**
I certify that I have:
- [X] tested my changes
- [x] added/updated automated tests
- [X] updated the changelog
- [X] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Comment by adam-wolfe** [comment]
Functionality LGTM. Will let others review.

File: packages/zosfiles/src/methods/copy/Copy.ts
Comment (traeok) [inline packages/zosfiles/src/methods/copy/Copy.ts:116]: The `dsorg` for an extended PDS is `PO-E`, so we'll need to update this line for detecting PDSEs:

```suggestion
            return dsorg === "PO-E" || dsorg === "PO";
```

**Review by traeok — CHANGES_REQUESTED** [review]
Thanks for your work on this Pujal! I left a comment about handling PDSEs, but other than that everything LGTM

File: packages/zosfiles/src/methods/copy/Copy.ts
Comment (zFernand0) [inline packages/zosfiles/src/methods/copy/Copy.ts:63]: I know we are focused on the APIs first, but I'm curious if we should update the corresponding help text on the CLI commands.

```shell

 EXAMPLES
 --------

   - Copy the data set named 'USER.FROM.SET' to the data set named 'USER.TO.SET':

      $ zowe zos-files copy data-set "USER.FROM.SET" "USER.TO.SET"

   - Copy the data set member named 'USER.FROM.SET(MEM1)' to the data set member named
   'USER.TO.SET(MEM2)':

      $ zowe zos-files copy data-set "USER.FROM.SET(mem1)" "USER.TO.SET(mem2)"

   - Copy the data set named 'USER.FROM.SET' to the data set member named 'USER.TO.SET(MEM2)':

      $ zowe zos-files copy data-set "USER.FROM.SET" "USER.TO.SET(mem2)"

   - Copy the data set member named 'USER.FROM.SET(MEM1)' to the data set named 'USER.TO.SET':

      $ zowe zos-files copy data-set "USER.FROM.SET(mem1)" "USER.TO.SET"

   - Copy the data set named 'USER.FROM.SET' to the data set named 'USER.TO.SET' and replace
   like-named members:

      $ zowe zos-files copy data-set "USER.FROM.SET" "USER.TO.SET" --replace

```

File: packages/zosfiles/src/methods/copy/Copy.ts
Comment (zFernand0) [inline packages/zosfiles/src/methods/copy/Copy.ts:150]: Should we add a unit test where the source PDS has no members?

And one more thing... would it be ok to return `success: true` here? 
Not a massive difference, but some user might feel like they did something wrong 😅 

The first one is with `success: false` and the second is `success: true`
image

**Review by zFernand0 — COMMENTED** [review]
will approve once Trae's comment is addressed 🙏

**Review by gejohnston — APPROVED** [review]
I asked a question about what kind of error the user might get. Unless the answer to that question surprises us horribly, I approve.

**Review by zFernand0 — COMMENTED** [review]
Thanks for updating the text help 🙏 

However, that caused a few snapshots to fail.
Also, there is a small merge conflict in the changelog file which needs to be addressed when updating from master 😋 

Once those items are addressed, I'm good to approve 🙏

File: packages/zosfiles/src/methods/copy/Copy.ts
Comment (zFernand0) [reply packages/zosfiles/src/methods/copy/Copy.ts:170]: Here is an example of how a `truncation error`, i.e. the source `lrecl` is greater than the target `lrecl`, might look like
image

---

And it will still upload the member, just that it will truncate the data. Then stops uploading members.
image

---

Curious if we could get some progress bars (ideally with the member name that's being copied) 😋 
That will give people a sense of how much might be left, or if a bad error occurs, they know where the copy operation stopped 😋

Comment (gejohnston) [reply packages/zosfiles/src/methods/copy/Copy.ts:170]: Might it be possible to add the name of the member to error during the catch block before we rethrow it? Maybe there are problems with this idea. Just asking that you evaluate the idea.

**Comment by pujal0909** [comment]
> I asked a question about what kind of error the user might get. Unless the answer to that question surprises us horribly, I approve.

**Review by anaxceron — APPROVED** [review]
Good changelogs @pujal0909!

File: packages/zosfiles/CHANGELOG.md
Comment (traeok) [inline packages/zosfiles/CHANGELOG.md:6]:
```suggestion
- Enhancement: The `Copy.dataset` method now recognizes partitioned data sets and can copy members of a source PDS into an existing target PDS. [#2386](https://github.com/zowe/zowe-cli/pull/2386)
```

**Review by traeok — APPROVED** [review]
Changes LGTM, thanks for addressing feedback! 😁

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

**Comment by zFernand0** [comment]
Labeling `release-current` until EoD in case other related PRs are merged 🙏

**Pull request merged**
